### PR TITLE
MAYA-114281 - Query a USD attributes locked state via getAttr

### PR DESCRIPTION
**Pull request by seando-adsk**
#### MAYA-114281 - Query a USD attributes locked state via getAttr
* Added tests for new Ufe support of attributes using Maya getAttr command.

### CODE REVIEW

File: test/lib/ufe/testAttribute.py [hunk 1]
```diff
@@ -135,6 +176,35 @@ def runUndoRedoUsingMayaSetAttr(self, attr, newVal, decimalPlaces=None):
         cmds.redo()
         self.assertEqual(attr.get(), newVal)
 
+    def runMayaGetAttrTest(self, ufeAttr, decimalPlaces=None):
+        # Not all versions of Maya support getAttr with Ufe attributes.
+        if not self.getAttrSupportsUfe:
+            return
```
Comment (seando-adsk) [inline test/lib/ufe/testAttribute.py:182]: Generic function used by different attribute type testing below. Will be silently skipped if getAttr does no support ufe.

File: test/lib/ufe/testAttribute.py [hunk 2]
```diff
@@ -192,7 +262,7 @@ def testAttributeGeneric(self):
         # Use our engine method to run the bulk of the test (all the stuff from
         # the Attribute base class). We use the xformOpOrder attribute which is
         # an unsupported USD type, so it will be a UFE Generic attribute.
-        ufeAttr, usdAttr = attrDict = self.runTestAttribute(
+        ufeAttr, usdAttr = self.runTestAttribute(
```
Comment (seando-adsk) [inline test/lib/ufe/testAttribute.py:265]: I noticed this extraneous attrDict var that I think was a mistake during the initial commit.

File: test/lib/ufe/testAttribute.py [hunk 3]
```diff
@@ -1051,18 +1174,46 @@ def testAttributeLock(self):
...
+            self.assertRaises(RuntimeError, cmds.setAttr, getsetAttrPath, currValue+1.0)
 
         # Test that we cannot set other Metadata again.
-        with self.assertRaises(RuntimeError):
-            attr.setMetadata('documentation', 'New Doc')
+        self.assertRaises(RuntimeError, attr.setMetadata, 'documentation', 'New Doc')
+
+    def testMayaGetAttr(self):
+        if not self.getAttrSupportsUfe:
+            self.skipTest('Maya getAttr command does not support Ufe')
```
Comment (seando-adsk) [inline test/lib/ufe/testAttribute.py:1193]: Specific getAttr test which will log message if skipped. Cannot use function decorator for reason indicated above.

File: test/lib/ufe/testAttribute.py [hunk 4]
```diff
@@ -1051,18 +1174,46 @@ def testAttributeLock(self):
...
+                               'Cannot evaluate more than one attribute\.$',
+                               cmds.getAttr,
+                               pathStr+'.xformOp:translate',
+                               pathStr+'.xformOpOrder')
+
+        # Mixing Maya and non-Maya attributes is an error.
+        self.assertRaisesRegex(RuntimeError,
+                               'Cannot evaluate more than one attribute\.$',
+                               cmds.getAttr,
+                               'proxyShape1.shareStage',pathStr+'.xformOp:translate')
```
Comment (seando-adsk) [inline test/lib/ufe/testAttribute.py:1216]: Test some of the expected error cases.

**Review by ppt-adsk — APPROVED** [review]

**Pull request merged**